### PR TITLE
Use dataclasses instead of namedtuples

### PR DESCRIPTION
**Pull request by grmnz**
I prepared a pr based on the comments from discussion https://github.com/aliev/aioauth/discussions/57

### CODE REVIEW

**Comment by aliev** [comment]
@grmnz sorry for the delay! checking now :)

File: tests/utils.py [hunk 1]
```diff
@@ -248,12 +305,12 @@
...
     """Converts dataclass object to dict and returns dict without empty values"""
-    return {key: value for key, value in query._asdict().items() if bool(value)}
+    return {key: value for key, value in asdict(query).items() if bool(value)}
 
 
 def set_values(model, values):
-    """Sets NamedTuple instance value and returns new NamedTuple"""
-    return model.__class__(**{**model._asdict(), **values})
+    """Return a new dataclass object replacing specified fields with new values"""
+    return replace(model, **values)
```
Comment (aliev) [inline tests/utils.py:313]: I would directly use replace instead of set_values and remove set_values function

**Review by aliev — CHANGES_REQUESTED** [review]

File: aioauth/models.py [hunk 2]
```diff
@@ -7,15 +7,16 @@
 
 ----
 """
-
+import dataclasses
```
Comment (aliev) [inline aioauth/models.py:10]: I think it looks shorter:

```suggestion
from dataclasses import dataclass
```

the same with other files

**Comment by aliev** [comment]
I removed Python 3.6 support within this PR: https://github.com/aliev/aioauth/pull/59 and updated this branch.

File: tests/utils.py (hunk 1 above)
Comment (grmnz) [reply tests/utils.py]: `set_values` was removed

**Review by aliev — APPROVED** [review]
Great job! Thanks for your contribution!

**Pull request merged**